felix/bpf-gpl/bpf.h

```diff
@@ -289,16 +271,17 @@ CALI_CONFIGURABLE_DEFINE(natin_idx, 0xdeadbeef)
 #define NATIN_IFACE	CALI_CONFIGURABLE(natin_idx)
 
 #ifdef UNITTEST
-CALI_CONFIGURABLE_DEFINE(__skb_mark, 0x4d424b53) /* be 0x4d424b53 = ASCII(SKBM) */
-#define SKB_MARK	CALI_CONFIGURABLE(__skb_mark)
-#endif
-
-#define MAP_PIN_GLOBAL	2
+#define CALI_PATCH_DEFINE(name, pattern)							\
+static CALI_BPF_INLINE __be32 cali_patch_##name()					\
+{												\
+	__u32 ret;										\
+	asm("%0 = " #pattern ";" : "=r"(ret) /* output */ : /* no inputs */ : /* no clobber */);\
+	return ret;										\
+}
+#define CALI_PATCH(name)	cali_patch_##name()
 
-#ifndef __BPFTOOL_LOADER__
-#define CALI_MAP_TC_EXT_PIN(pin)	.pinning_strategy = pin,
-#else
-#define CALI_MAP_TC_EXT_PIN(pin)
+CALI_PATCH_DEFINE(__skb_mark, 0x4d424b53) /* be 0x4d424b53 = ASCII(SKBM) */
+#define SKB_MARK	CALI_PATCH(__skb_mark)
 #endif
 
 #define map_symbol(name, ver) name##ver
```

Comment:
Nit QQ: should `map_symbol` be uppercase consistent with other `#define` statements?

Reply:
some defines are meant to look more like functions